test/parallel/test-writefloat.js

```diff
@@ -26,110 +26,118 @@
+assert.strictEqual(0x00, buffer[2]);
+assert.strictEqual(0x00, buffer[3]);
+assert.strictEqual(0x00, buffer[4]);
+assert.strictEqual(0x00, buffer[5]);
+assert.strictEqual(0x80, buffer[6]);
+assert.strictEqual(0x7F, buffer[7]);
+assert.strictEqual(Infinity, buffer.readFloatBE(0));
+assert.strictEqual(Infinity, buffer.readFloatLE(4));
 
-  buffer.writeFloatBE(-Infinity, 0);
-  buffer.writeFloatLE(-Infinity, 4);
-  // Darwin ia32 does the other kind of NaN.
-  // Compiler bug.  No one really cares.
-  assert(0xFF === buffer[0] || 0x7F === buffer[0]);
-  assert.strictEqual(0x80, buffer[1]);
-  assert.strictEqual(0x00, buffer[2]);
-  assert.strictEqual(0x00, buffer[3]);
-  assert.strictEqual(0x00, buffer[4]);
-  assert.strictEqual(0x00, buffer[5]);
-  assert.strictEqual(0x80, buffer[6]);
-  assert.strictEqual(0xFF, buffer[7]);
-  assert.strictEqual(-Infinity, buffer.readFloatBE(0));
-  assert.strictEqual(-Infinity, buffer.readFloatLE(4));
+buffer.writeFloatBE(-Infinity, 0);
+buffer.writeFloatLE(-Infinity, 4);
+// Darwin ia32 does the other kind of NaN.
+// Compiler bug.  No one really cares.
+assert(0xFF === buffer[0] || 0x7F === buffer[0]);
+assert.strictEqual(0x80, buffer[1]);
+assert.strictEqual(0x00, buffer[2]);
+assert.strictEqual(0x00, buffer[3]);
+assert.strictEqual(0x00, buffer[4]);
+assert.strictEqual(0x00, buffer[5]);
+assert.strictEqual(0x80, buffer[6]);
+assert.strictEqual(0xFF, buffer[7]);
+assert.strictEqual(-Infinity, buffer.readFloatBE(0));
+assert.strictEqual(-Infinity, buffer.readFloatLE(4));
 
-  buffer.writeFloatBE(NaN, 0);
-  buffer.writeFloatLE(NaN, 4);
-  // Darwin ia32 does the other kind of NaN.
-  // Compiler bug.  No one really cares.
-  assert(0x7F === buffer[0] || 0xFF === buffer[0]);
-  // mips processors use a slightly different NaN
-  assert(0xC0 === buffer[1] || 0xBF === buffer[1]);
-  assert(0x00 === buffer[2] || 0xFF === buffer[2]);
-  assert(0x00 === buffer[3] || 0xFF === buffer[3]);
-  assert(0x00 === buffer[4] || 0xFF === buffer[4]);
-  assert(0x00 === buffer[5] || 0xFF === buffer[5]);
-  assert(0xC0 === buffer[6] || 0xBF === buffer[6]);
-  // Darwin ia32 does the other kind of NaN.
-  // Compiler bug.  No one really cares.
-  assert(0x7F === buffer[7] || 0xFF === buffer[7]);
-  assert.ok(Number.isNaN(buffer.readFloatBE(0)));
-  assert.ok(Number.isNaN(buffer.readFloatLE(4)));
-}
+buffer.writeFloatBE(NaN, 0);
+buffer.writeFloatLE(NaN, 4);
+// Darwin ia32 does the other kind of NaN.
+// Compiler bug.  No one really cares.
+assert(0x7F === buffer[0] || 0xFF === buffer[0]);
+// mips processors use a slightly different NaN
+assert(0xC0 === buffer[1] || 0xBF === buffer[1]);
+assert(0x00 === buffer[2] || 0xFF === buffer[2]);
+assert(0x00 === buffer[3] || 0xFF === buffer[3]);
+assert(0x00 === buffer[4] || 0xFF === buffer[4]);
+assert(0x00 === buffer[5] || 0xFF === buffer[5]);
+assert(0xC0 === buffer[6] || 0xBF === buffer[6]);
+// Darwin ia32 does the other kind of NaN.
+// Compiler bug.  No one really cares.
```

Comment:
I think these comments (and checks) are obsolete now since V8 canonicalizes nans.  There might still be platform differences but not compiler differences.

(Likewise in test/parallel/test-writedouble.js.)

Reply:
I removed them and the corresponding extra checks. I also reworked the existing tests to be much smaller and renamed the files as they actually miss the buffer subsystem in their name and added more OOB tests.